public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts

```diff
@@ -142,48 +146,58 @@ export default class Datasource extends DataSourceApi<AzureMonitorQuery, AzureDa
     return this.azureLogAnalyticsDatasource.annotationQuery(options);
   }
 
-  async testDatasource(): Promise<DatasourceValidationResult> {
-    const promises: Array<Promise<DatasourceValidationResult>> = [];
+  private isValidConfigField(field?: string): boolean {
+    return typeof field === 'string' && field.length > 0;
+  }
 
-    promises.push(this.azureMonitorDatasource.testDatasource());
-    promises.push(this.azureLogAnalyticsDatasource.testDatasource());
+  private validateDatasource(): DatasourceValidationResult | undefined {
+    const authType = getAuthType(this.instanceSettings);
 
-    return await Promise.all(promises).then((results) => {
-      let status: 'success' | 'error' = 'success';
-      let message = '';
+    if (authType === 'clientsecret') {
+      if (!this.isValidConfigField(this.instanceSettings.jsonData.tenantId)) {
+        return {
+          status: 'error',
+          message: 'The Tenant Id field is required.',
+        };
+      }
```

Comment:
I kept this from the original `testDatasource` methods. Setting the field to `required` seems to have no effect. Unsure if that's because they're using legacy forms (should probably be updated as well). The error handling could be improved to provide the appropriate information though I think the form should be updated to use the newer components where the `required` prop should have an effect..

Reply:
> though I think the form should be updated to use the newer components where the `required` prop should have an effect

I agree but in the meantime, I would remove this code. It's not worth it to keep a reference to `instanceSettings` and everything just to validate a mandatory input (which is obvious anyway).